client/signup/steps/website-content/page-details.tsx

```diff
@@ -0,0 +1,100 @@
+import { useTranslate } from 'i18n-calypso';
+import { ChangeEvent } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import {
+	SubLabel,
+	Label,
+	TextAreaField,
+	HorizontalGrid,
+} from 'calypso/signup/accordion-form/form-components';
+import { imageUploaded, textChanged } from 'calypso/state/signup/steps/website-content/actions';
+import { PageData } from 'calypso/state/signup/steps/website-content/schema';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { SiteData } from 'calypso/state/ui/selectors/get-selected-site';
+import { MediaUploadData, WordpressMediaUpload } from './wordpress-media-upload';
+
+export function PageDetails( {
+	page,
+	formErrors,
+	onChangeField,
+}: {
+	page: PageData;
+	formErrors: any;
+	onChangeField?: (
+		{
+			target: { name, value },
+		}: ChangeEvent< HTMLInputElement >,
+		pageTitle: string
+	) => void;
+} ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const site = useSelector( getSelectedSite );
+	const pageTitle = page.title;
+
+	const onMediaUploaded = ( { title, URL, uploadID, mediaIndex }: MediaUploadData ) => {
+		dispatch(
+			imageUploaded( {
+				title: page.title,
+				image: { url: URL, caption: title, uploadID },
+				mediaIndex,
+			} )
+		);
+		onChangeField &&
+			onChangeField(
+				{
+					target: { name: 'image', value: URL },
+				} as ChangeEvent< HTMLInputElement >,
+				pageTitle
+			);
+	};
+
+	const onContentChange = ( e: ChangeEvent< HTMLInputElement > ) => {
+		const {
+			target: { value },
+		} = e;
+		dispatch(
```

Comment:
Decided to directly push to redux on form change. This way, whatever is typed is persisted instantly. We can probably add a debounce in the future if the UX becomes slow, but that is highly unlikely I think!

Reply:
This component will not be reusable since it is now tied to the redux state. Can we make this a dumb component and handle state (redux/local) higher up the component tree?

Reply:
I did not model this as a reusable component actually. Not sure how this can be reusab. Do you see this being reused elsewhere?